src/middleware.ts

```diff
@@ -106,3 +107,84 @@ export const withWebhookSignatureVerified =
 
     return handler(request);
   };
+
+export interface DashboardTokenPayload extends JwtPayload {
+  app: string;
+}
+
+export const withJWTVerified =
+  (getAppId: (request: Request) => Promise<string | undefined>): Middleware =>
+  (handler) =>
+  async (request) => {
+    const { [SALEOR_DOMAIN_HEADER]: saleorDomain, "authorization-bearer": token } = request.headers;
+
+    if (token === undefined) {
+      return Response.BadRequest({
+        success: false,
+        message: "Missing token.",
+      });
+    }
+
+    let tokenClaims;
+    try {
+      tokenClaims = jwt.decode(token as string);
+    } catch (e) {
+      console.error(e);
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    if (tokenClaims === null) {
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    if ((tokenClaims as DashboardTokenPayload).iss !== saleorDomain) {
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    let appId: string | undefined;
+    try {
+      appId = await getAppId(request);
+    } catch (error) {
+      console.error("Error during getting the app ID.");
+      console.error(error);
+      return Response.BadRequest({
+        success: false,
+        message: "Error during token invalidation - could not obtain the app ID.",
+      });
+    }
+
+    if (!appId || (tokenClaims as DashboardTokenPayload).app !== appId) {
+      return Response.BadRequest({
+        success: false,
+        message: "Invalid token.",
+      });
+    }
+
+    const jwksClient = jwks({
+      jwksUri: `https://${saleorDomain}/.well-known/jwks.json`,
```

Comment:
this should use https://github.com/saleor/saleor-app-sdk/blob/main/src/urls.ts#L1 for determining whether to use `http` or `https`. Otherwise you will get some obscure error when trying to debug/develop using locally deployed sailor under http://localhost:8000 (I know, that in template it's that that way, I had just at the top of my personal backlog to take care of that, when moving to SDK).